pkg/drain/nodegroup.go

```diff
@@ -143,32 +158,37 @@ func (n *NodeGroupDrainer) Drain() error {
 
 			logger.Debug("already drained: %v", mapToList(drainedNodes.Items()))
 			logger.Debug("will drain: %v", newPendingNodes.List())
-			for i, node := range newPendingNodes.List() {
-				if err := sem.Acquire(ctx, 1); err != nil {
-					logger.Critical("failed to acquire semaphore: %w", err)
-				}
 
-				go func(i int, node string) {
+			errorGroup, _ := errgroup.WithContext(ctx)
```

Comment:
```suggestion
			g, _ := errgroup.WithContext(ctx)
```

Reply:
We should also use the context here so `Wait` can cancel it when it's done so there are no stragglers.